src/cargo_communication/request_actor.rs

```diff
@@ -348,3 +348,837 @@ pub trait CargoHandleTrait<T> {
 
     fn join(self) -> io::Result<ExitStatus>;
 }
+
+#[cfg(test)]
+pub mod compile_request_tests {
+    use std::collections::HashSet;
+    use std::fmt::Debug;
+    use std::hash::Hash;
+    use std::os::unix::prelude::ExitStatusExt;
```

Comment:
dobra, to już natychmiast budzi moje obawy zanim przeczytałem sam kod: co z windowsem?

Reply:
Poprawiłem